### ISSUE TITLE
All locations searched show "Weak or No WiFi connection"

### ISSUE THREAD

**Issue by catdogmat** [bmaroti9/Overmorrow#46]
Hi,
When I search any location and try to view the weather, it takes a long time to load, then shows a weak or no WiFi connection message. I currently have perfectly fine WiFi. The default of "New York" loads fine as well.
Thanks!

**Comment by catdogmat** [comment]
I tried to change the search provider to open metro and it works slightly better. I can now search, but using my current location, I get the same error.

**Comment by catdogmat** [comment]
Interesting. With some more testing, I've found that some search terms are fine but some error! Maybe it's a rate limit of sorts?

**Comment by catdogmat** [comment]
Awesome, thanks. Do you think it would be a good idea to have an easy way to allow people to use their own API keys? That could help solve the rate limiting problem.

**Comment by bmaroti9** [comment]
Yeah, so i guess i could do that, for some providers, but for example if you look at open-meteo or met-norway neither have any api keys.

**Comment by bmaroti9** [comment]
Hey, just curious if you're still getting this. I have noticed the same sometimes, but only with weatherapi and met norway. I have traced it down to a timeapi.io call, and it seems like their api is kind of lagging and sometimes just doesn't return anything.

**Comment by bmaroti9** [comment]
I have done a lot of improvements on this in v2.4.6! Hopefully this fixes it.
